tests/test_pipelines.py

```diff
@@ -155,3 +166,57 @@ def test_pa11y_not_installed(mocker):
         ["node_modules/.bin/pa11y", "--version"],
         stdout=DEVNULL, stderr=DEVNULL,
     )
+
+
+def test_pa11y_title_mismatch(mocker, tmpdir):
+    item = {
+        "url": "http://courses.edx.org/ponies",
+        "page_title": "Sparkly Ponies of Joy",
+        "request_headers": {"Cookie": "nocookieforyou"},
+        "accessed_at": datetime(2016, 8, 20, 14, 12, 45),
+    }
+    fake_pa11y_data = {
+        "results": [{
+            "message": "Check that the title element describes the document.",
+            "code": "WCAG2AA.Principle2.Guideline2_4.2_4_2.H25.2",
+            "type": "notice",
+            "html": "<title>Evil Demons of Despa...</title>"
+        }]
+    }
+
+    # setup
+    data_dir = tmpdir.mkdir("data")
+    spider = mocker.Mock(data_dir=str(data_dir))
+
+    # fake subprocess
+    def popen_side_effect(args, *pos_args, **kwargs):
+        if "--version" in args:
+            version_process = mocker.Mock(returncode=None)
+            def mock_wait():
+                version_process.returncode = 0
+            version_process.wait.side_effect = mock_wait
+            return version_process
+        else:
+            run_process = mocker.Mock(returncode=None)
+            def mock_communicate():
+                run_process.returncode = 2
+                # returns both stdout and stderr
+                return json.dumps(fake_pa11y_data), ""
+            run_process.communicate.side_effect = mock_communicate
+            return run_process
+    mocker.patch("subprocess.Popen", side_effect=popen_side_effect)
+
+    # stub out config file creation/removal
+    mocker.patch("tempfile.NamedTemporaryFile")
```

Comment:
Nope! [pytest-mock](https://github.com/pytest-dev/pytest-mock/) handles that for you, as well as undoing the mocks at the end of the test.